paddle/fluid/operators/one_hot_op.cc

```diff
@@ -86,6 +86,11 @@ class OneHotOpMaker : public framework::OpProtoAndCheckerMaker {
                  "An integer to specify the data type of one-hot "
                  "vector. The default value is FP32.")
         .SetDefault(paddle::framework::proto::VarType::FP32);
+    AddAttr<bool>("allow_out_of_range",
```

Comment:
Why add this? Is it better to report error when out of bound occurs?

Reply:
we added distributed fc support which separates the fc parameter to different gpu cards, and it collaborates with labels to calculate loss. In our case, the label should firstly recalculated by trainer numbers, and then change to one hot representation, e.g.
two trainer: t1, t2, with data parallel
class dimension: 10, and each trainer takes 5, i.e. we divided the whole class dimension into 2.
batched label in t1: 2, 4, 6, 8
batched label in t2: 2, 4, 6, 8 (the same with t1 for simple illustration)
firstly, label is recalculated to
t1: 2, 4, -1, -1
t2: -1, -1, 1, 3, in which, (6-5)%5=1, (8-5)%5=3
then we get one hot representation as
t1: 00100,00001,00000,00000
t2: 00000,00000,01000,00100
u see, we need correctly convert `out_of_range` label -1 to uniform 0 representation, that why we add this api

Reply:
I see. Thanks!